grafana/resource_cloud_stack.go

```diff
@@ -268,3 +287,28 @@ func FlattenStack(d *schema.ResourceData, stack gapi.Stack) {
 	d.Set("alertmanager_url", stack.AmInstanceURL)
 	d.Set("alertmanager_status", stack.AmInstanceStatus)
 }
+
+// waitForStackReadiness retries until the stack is ready, verified by querying the Grafana URL
+func waitForStackReadiness(ctx context.Context, d *schema.ResourceData) diag.Diagnostics {
+	if wait := d.Get("wait_for_readiness").(bool); !wait {
+		return nil
+	}
+
+	err := resource.RetryContext(ctx, 2*time.Minute, func() *resource.RetryError {
+		resp, err := http.Head(d.Get("url").(string))
```

Comment:
I think you should utilize the given parent `ctx` context to fulfills possible timeouts/cancellations during the HTTP request.